test/integration/test_backend.py

```diff
@@ -106,8 +106,7 @@ def test_sim_backend_options(self):
         provider: IBMProvider = self.backend.provider
         backend = provider.get_backend("ibmq_qasm_simulator")
         backend.options.shots = 2048
-        backend.set_options(memory=True)
-        job = backend.run(ReferenceCircuits.bell(), shots=1024, foo="foo")
+        job = backend.run(ReferenceCircuits.bell(), shots=1024, memory=True, foo="foo")
```

Comment:
Why the change? Shouldn't `test_sim_backend_options` include setting options?

Reply:
this test is [failing](https://github.com/Qiskit/qiskit-ibm-provider/actions/runs/3383220181/jobs/5618919642#step:5:687) because in `_get_run_config` memory will default to False if it's not directly passed into `backend.run()`.  

`set_options(memory=True)` won't actually set the memory flag to True

Reply:
> set_options(memory=True) won't actually set the memory flag to True when running a job

Looks like the test found a bug! The parent `BackendV2` class says you can update the option using `set_options`. `_get_run_config` should have merged `self. _options` with `kwargs`.